### PR TITLE
feat: ssg article

### PR DESCRIPTION
**Pull request by joshua-mo-143**
todo:
thumbnail image
seo/promo plan 
redo frontmatter

### CODE REVIEW

File: _blog/2023-11-15-ssg-in-rust.mdx [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+html {
+    display: flex;
+    text-align: center;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    width: 90%;
+}
+```
+As you can see, it's not quite Awwwards-worthy CSS. However, it'll do for making our pages look at least somewhat aligned.
```
Comment (jonaro00) [inline _blog/2023-11-15-ssg-in-rust.mdx:74]: typo or intended?

File: _blog/2023-11-15-ssg-in-rust.mdx [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+    html_output.push_str("<head>"); 
+    html_output.push_str( 
+        r#"<meta charset="UTF-8"/> 
+    <meta name="viewport" content="width=device-width, initial-scale=1.0"/> 
+    <meta http-equiv="X-UA-Compatible" content="ie=edge"/> 
+    <link rel="stylesheet" type="text/css" href="/styles.css">"#); 
+    html_output.push_str(&format!( 
+        "<title>{title}</title> 
+        <meta property=\"og:title\" content=\"{title}\"/> 
+        <meta property=\"og:description\" content=\"{description}\"/>"
```
Comment (jonaro00) [inline _blog/2023-11-15-ssg-in-rust.mdx:123]: This one could have `r#` as well IIRC

File: _blog/2023-11-15-ssg-in-rust.mdx [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+        Ok(res) => res,
+        Err(_) => {
+            file = format!("templates/{page}/index.md");
+            match tokio::fs::read(file).await {
+                Ok(res) => res,
+                Err(_) => return Html("Couldn't find this page - does it exist?".to_string()),
+            }
+        }
+    };
+// .. rest of code
```
Comment (jonaro00) [inline _blog/2023-11-15-ssg-in-rust.mdx:216]: Can indent

File: _blog/2023-11-15-ssg-in-rust.mdx [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+}
+```
+
+I spun up the local server and checked to see if it works or not - it does! I had previously created an index file and the `hello_world` route appear on the navbar. However, the links have no space between each other - which is bad. I made a quick, small addition to the styling file so the site links aren't running headlong into each other:
+```css
+// templates/styles.css
+#nav {
+display: flex;
+justify-content: center;
+gap: 2em;
```
Comment (jonaro00) [inline _blog/2023-11-15-ssg-in-rust.mdx:307]: ditto

**Review by jonaro00 — APPROVED** [review]

File: _blog/2023-11-15-ssg-in-rust.mdx (hunk 1 above)
Comment (joshua-mo-143) [reply _blog/2023-11-15-ssg-in-rust.mdx:74]: Oh, maybe I should have a link to what Awwwards is there.

Awwwards is a website for showcasing the best looking websites and they have competitions regularly

File: _blog/2023-11-15-ssg-in-rust.mdx (hunk 2 above)
Comment (joshua-mo-143) [reply _blog/2023-11-15-ssg-in-rust.mdx:123]: I am not sure about this, do `r#` strings allow variable printing?

**Pull request merged**